internal/conversion/nodeselector.go

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier:Apache-2.0
+
+package conversion
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+
+	"github.com/openperouter/openperouter/api/v1alpha1"
+)
+
+// NodeMatchesSelector returns true if the given node matches the label selector.
+// If the selector is nil or empty, it matches all nodes (backward compatible).
+func NodeMatchesSelector(node *corev1.Node, selector *metav1.LabelSelector) (bool, error) {
```

Comment:
this is a nit. 
please bear with me for a sec ... 

I really dislike func signatures returning `(bool, error)`. I think they're confusing. 

I think we can simplify the func's signature easily in this case; how about we receive as parameter the selector ? i.e. the output of `metav1.LabelSelectorAsSelector(selector)` ? That's the only thing that returns an error in this function. 

You can disregard this comment if you think it's not that bad.